File: assignments/integer_to_roman/python/integer_to_roman_test_runner.py
import sys
from integer_to_roman import IntegerToRoman

def run_tests():
    solver = IntegerToRoman()

    test_cases = [
        (3, "III"),
        (4, "IV"),
        (9, "IX"),
        (58, "LVIII"),
        (1994, "MCMXCIV"),
        (3749, "MMMDCCXLIX"),
        (3999, "MMMCMXCIX"),
    ]

    passed_count = 0
    report = []

    print("===== Starting Automated Tests for IntegerToRoman (Python) =====")

    for i, (input_val, expected_output) in enumerate(test_cases):
        try:
            actual_output = solver.convert(input_val)

            report.append(f"Test Case #{i + 1}: Input '{input_val}'")
            report.append(f" -> Expected: \"{expected_output}\", Got: \"{actual_output}\"")

            if actual_output == expected_output:
                report.append(" -> ✅ PASS\n")
                passed_count += 1
            else:
                report.append(" -> ❌ FAIL\n")

        except Exception as e:
            report.append(f"Test Case #{i + 1}: Input '{input_val}'")
            report.append(f" -> ❌ ERROR: An exception occurred - {type(e).__name__}\n")

    print("\n".join(report))
    print("===== Test Summary =====")
    print(f"Total Tests: {len(test_cases)}")
    print(f"Passed: {passed_count}")
    print(f"Failed: {len(test_cases) - passed_count}")
    print("========================")

    print(f"[SUMMARY]:{passed_count}:{len(test_cases)}")

if __name__ == "__main__":
    sys.path.append('.')
    run_tests()

File: assignments/integer_to_roman/javascript/integer_to_roman_test_runner.js
const assert = require('assert');
const { IntegerToRoman } = require('./IntegerToRoman.js');

function runTests() {
    const solver = new IntegerToRoman();
    const testCases = [
        { input: 3, expected: "III" }, { input: 4, expected: "IV" }, { input: 9, expected: "IX" },
        { input: 58, expected: "LVIII" }, { input: 1994, expected: "MCMXCIV" }, { input: 3749, expected: "MMMDCCXLIX" }
    ];

    let passedCount = 0;
    let report = [];
    console.log("===== Starting Automated Tests for IntegerToRoman (JavaScript) =====");

    testCases.forEach((tc, i) => {
        try {
            const actual = solver.convert(tc.input);
            assert.strictEqual(actual, tc.expected);
            report.push(`Test Case #${i + 1}: Input ${tc.input} -> ✅ PASS`);
            passedCount++;
        } catch (error) {
            report.push(`Test Case #${i + 1}: Input ${tc.input} -> ❌ FAIL/ERROR: Expected "${tc.expected}", but got "${error.actual}"`);
        }
    });

    console.log(report.join('\n\n'));
    console.log("\n===== Test Summary =====");
    console.log(`Total Tests: ${testCases.length}`); console.log(`Passed: ${passedCount}`); console.log(`Failed: ${testCases.length - passedCount}`); console.log("========================");
    console.log(`[SUMMARY]:${passedCount}:${testCases.length}`);
}

runTests();